pkg/providers/vsphere/reconciler/reconciler_test.go

```diff
@@ -149,6 +149,17 @@ func TestReconcileCNIErrorClientRegistry(t *testing.T) {
 	tt.Expect(result).To(Equal(controller.Result{}))
 }
 
+func TestReconcilerReconcileControlPlaneSuccess(t *testing.T) {
+	tt := newReconcilerTest(t)
+	tt.createAllObjs()
+
+	result, err := tt.reconciler().ReconcileControlPlane(tt.ctx, test.NewNullLogger(), tt.buildSpec())
+
+	tt.Expect(err).NotTo(HaveOccurred())
+	tt.Expect(tt.cluster.Status.FailureMessage).To(BeZero())
+	tt.Expect(result).To(Equal(controller.Result{}))
```

Comment:
I would be great if we could check that the capi objects for workers have been created :)

Reply:
You mean check if the control plane objects are created during the test right? How do you capture a variable created in runtime without return it

Reply:
Yeah exactly that
Ahh great question
You can use the client.Client to read it from the cluster :)